system/lib/wasmfs/memory_backend.h

```diff
@@ -18,6 +18,8 @@ namespace wasmfs {
 class MemoryFile : public DataFile {
   std::vector<uint8_t> buffer;
 
+  void open() override {}
+  void close() override {}
```

Comment:
Maybe just male these default to empty function rather than pure virtual?  (seems like overriding them is optional).

Reply:
Actually, I expect that all nontrivial backends will need to override these to get the correct behavior for accessing files after they have been unlinked, so I think it would be better to keep these non-optional.